app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt

```diff
@@ -31,35 +32,54 @@ class HomeMenu(
 
     val menuBuilder by lazy { BrowserMenuBuilder(menuItems) }
 
+    private val hasAccountProblem = context.components.backgroundServices.accountManager.accountNeedsReauth()
```

Comment:
Does this need to be a getter? When is `HomeMenu` recreated?

Reply:
The sync state might change. The Home menu, is recreated in `onViewCreated` in `HomeFragment`

Reply:
Since the sync state may change this should probably be a getter.
```suggestion
    private val hasAccountProblem get() = context.components.backgroundServices.accountManager.accountNeedsReauth()
```